src/clients/go/pkg/types/main.go

```diff
@@ -42,12 +42,12 @@ func (value Uint128) String() string {
 	return s[lastNonZero:]
 }
 
-func (value Uint128) BigInt() big.Int {
+func (value Uint128) BigInt() *big.Int {
 	// big.Int uses bytes in big-endian but Uint128 stores bytes in little endian, so reverse it.
 	bytes := value.Bytes()
 	swapEndian(bytes[:])
 
-	ret := big.Int{}
+	ret := &big.Int{}
 	ret.SetBytes(bytes[:])
 	return ret
```

Comment:
The code here is being GC-conscious by avoiding the `big.Int` itself being heap allocated. In this case I think users could do `&u128.BigInt()` instead.

Reply:
#TIL thank you